ubyssey/static/src/js/components/Search.jsx

```diff
@@ -37,36 +38,33 @@ var Search = React.createClass({
             if (this.state.cache.has(q)) {
                 this.setState({ results: this.state.cache.get(q) });
             } else {
-                dispatch.search('article', {q: q}, function(data){
-                    this.setState(function(prevState, props){
+                dispatch.search('article', { q }, data => {
+                    this.setState(prevState => {
                         prevState.results = data.results;
                         prevState.cache.set(q, data.results);
                         return prevState;
                     });
-                }.bind(this));
+                });
             }
         } else {
             this.setState({ results: [] });
         }
     },
-    render: function(){
-
-        var results = this.state.results.map(function(item, i){
-            return (
-                <li><a href={item.url} >{item.headline}</a></li>
-            );
-        });
+    render() {
+        const results = this.state.results.map((item, i) => (
+            <li key={i}><a href={item.url}>{item.headline}</a></li>
+        ));
 
         return (
             <div className="u-container">
                 <form method="get" action="/archive/">
-                    <label for="author-search"><i className="fa fa-search"></i></label>
-                    <input ref="search" className={this.state.results.length > 0 ? "open" : ""} name="q" id="search-bar" type="text" autoComplete="off" onChange={this.updateQuery} value={this.state.q} placeholder="Search The Ubyssey..." />
+                    <label htmlFor="author-search"><i className="fa fa-search"></i></label>
+                    <input ref="search" className={this.state.results.length > 0 && "open"} name="q" id="search-bar" type="text" autoComplete="off" onChange={this.updateQuery} value={this.state.q} placeholder="Search The Ubyssey..." />
                 </form>
-                <ul className={"results" + (this.state.results.length > 0 ? " open" : "")}>{results}</ul>
+                <ul className={"results" + (this.state.results.length > 0 && " open")}>{results}</ul>
```

Comment:
`(this.state.results.length > 0 && " open")`

This appends "false" to the className when `results.length == 0`.